pkgs/development/python-modules/jinja2/2.nix

```diff
@@ -23,8 +23,14 @@ buildPythonPackage rec {
   # See https://github.com/pallets/jinja/issues/1158
   doCheck = !stdenv.is32bit || isPy3k;
 
-  checkPhase = ''
-    pytest -v tests -W ignore::ResourceWarning -W ignore::DeprecationWarning
+  checkPhase = let
+    pytestArgs = lib.optionals isPy3k [
+      "-W ignore::ResourceWarning"
+    ] ++ [
+      "-W ignore::DeprecationWarning"
+    ];
+  in ''
+    pytest -v tests ${toString pytestArgs}
```

Comment:
pytestCheckHook would be a bit cleaner to use and we could save half of this construct.